eth/roundinitializer.go

```diff
@@ -103,6 +107,22 @@ func (r *RoundInitializer) tryInitialize() error {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
+	if r.tw.LastSeenL1Block().Cmp(r.nextRoundStartL1Block) < 0 {
+		// Round already initialized
+		return nil
+	}
+
+	if r.maxDelay > 0 {
+		randDelay := time.Duration(rand.Int63n(int64(r.maxDelay)))
+		glog.Infof("Waiting %v before attempting to initialize round", randDelay)
+		time.Sleep(randDelay)
```

Comment:
Is there a risk that L89 would be triggered multiple times as we're sleeping here? IIUC, `l1BlockSink` will stream every new block seen on the chain, so if we wait here up to 30s we could have multiple blocks showing up and multiple `tryInitialize` functions executing (likely sleeping) on the same node. 

If that's the case, wdyt of adding some kind of mutexing here? One built-in solution could be a [singleflight](https://pkg.go.dev/golang.org/x/sync/singleflight) but maybe a manual mutex might be simpler.

Reply:
Hmm, or maybe simpler could also be to have the sleep logic outside the go-routine, blocking the main watch loop. Not sure, will leave it up to you!

Reply:
There is a mutex in L107, so the worst that can happen is that while waiting we'll create a few unnecessary goroutines. But that should be ok.